lambdas/passwordRenewal/src/main/java/uk/gov/di/ipv/cri/drivingpermit/event/handler/PasswordRenewalHandler.java

```diff
@@ -0,0 +1,238 @@
+
+    @ExcludeFromGeneratedCoverageReport
+    public PasswordRenewalHandler() {
+        secretsManagerClient =
+                SecretsManagerClient.builder()
+                        .region(Region.EU_WEST_2)
+                        .credentialsProvider(EnvironmentVariableCredentialsProvider.create())
+                        .build();
+        ParameterStoreService parameterStoreService =
+                new ParameterStoreService((ParamManager.getSsmProvider()));
+        ObjectMapper objectMapper = new ObjectMapper().registerModule(new JavaTimeModule());
+        eventProbe = new EventProbe();
+        HttpRetryer httpRetryer = new HttpRetryer(HttpClients.custom().build(), eventProbe, 0);
+        DvlaConfiguration dvlaConfiguration = new DvlaConfiguration(parameterStoreService);
+        RequestConfig defaultRequestConfig = new HttpRequestConfig().getDefaultRequestConfig();
+        changePasswordService =
+                new ChangePasswordService(
+                        dvlaConfiguration,
+                        httpRetryer,
+                        defaultRequestConfig,
+                        objectMapper,
+                        eventProbe);
+        tokenRequestService =
+                new TokenRequestService(
+                        dvlaConfiguration,
+                        DataStore.getClient(),
+                        httpRetryer,
+                        defaultRequestConfig,
+                        objectMapper,
+                        eventProbe);
+    }
+
+    public PasswordRenewalHandler(
+            SecretsManagerClient secretsManagerClient,
+            ChangePasswordService changePasswordService,
+            TokenRequestService tokenRequestService,
+            EventProbe eventProbe) {
+        this.secretsManagerClient = secretsManagerClient;
+        this.changePasswordService = changePasswordService;
+        this.tokenRequestService = tokenRequestService;
+        this.eventProbe = eventProbe;
+    }
+
+    @Logging
+    @Metrics(captureColdStart = true)
+    public Void handleRequest(SecretsManagerRotationEvent input, Context context) {
+        LOGGER.info("step {} secretId: {}", input.getStep(), input.getSecretId());
+
+        SecretsManagerRotationStep.of(input.getStep())
+                .ifPresent(
+                        step -> {
+                            try {
+                                if (step == SecretsManagerRotationStep.CREATE_SECRET) {
+                                    /*First step is for the new password to be generated, which will be labelled as 'AWSPENDING'
+                                     * and stored in the Secrets Manager as the pending new password until it's been verified by DVLA*/
+                                    LOGGER.info("Generating a new password");
+                                    String newPassword = generateNewPassword();
+
+                                    if (null
+                                            == getValue(
```

Comment:
I think it would be worthing logging here that we have entered the if statement and place a value into secrets manager